File: archive/question_papers/JEE/2016/python_comp/codes/ee16b1030.m
clear;
clc;
close;

function y = f(x)
y = abs(log(2)-sin(x));
endfunction

function y = g(x)
y = f(f(x));
endfunction


x = linspace(-1,1,100);
plot(x,g(x),'LineWidth',4)
grid
xlabel('x')
ylabel('g(x)')

print('ee16b1030.eps','-color');

h=10.^(-10);
%right hand limit
(g(h)-g(0))/h
%left hand limit
(g(0)-g(-h))/h
%actual limit
cos(log(2))

File: archive/question_papers/JEE/2016/python_comp/codes/ee16b1030.py
import numpy as np
import matplotlib.pyplot as plt

def f(x):
    return np.abs(np.log(2) - np.sin(x))
def g(x):
    return f(f(x))

x = np.linspace(-1,1,100)

plt.plot(x,g(x))
plt.grid()
plt.xlabel('$x$')
plt.ylabel('$g(x)$')
plt.savefig('../figs/ee16b1030.eps')
plt.show()

h = 10**(-10)

#right hand limit
print (g(h) - g(0))/h
     
#left hand limit
print (g(0) - g(-h))/h

print np.cos(np.log(2))
